### DISCUSSION TITLE
Windows Coreminer Setup Instructions and Guide for Beginners

### DISCUSSION THREAD

**Discussion by tgh90** [catchthatrabbit/coreminer#27] [Operating System]
### Windows Coreminer Setup Instructions and Guide for Beginners

1. Open the Microsoft Store and search for **Ubuntu 22.04.2 LTS** and install it. It is a Virtual Machine and Command Line Interface (CLI) of the operating system Ubuntu (Linux) which you'll need to get the miner running. Open the virtual machine **Ubuntu 22.04.2 LTS** and wait a few seconds. Now you have to enter a new UNIX username. Just use only lowercase letters and it will work (e.g. coreminer). Then type in a password and you are ready to go.

2. You're not in a Graphical User Interface (GUI) anymore and you control the CLI with commands. To see in which directory you are currently in type in `la` or `ls`. If nothing is shown there is no folder yet in the current directory. Every command I am showing in this guide is one line even though it could happen that it is shown in two lines here on GitHub.

3. Create a new folder: `mkdir coreminer`

4. Change directory: `cd coreminer`

5. Download the Coreminer software. The current version used in this guide is v0.19.80.
    For a x86_64 CPU it's:
    `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.80/coreminer-linux-x86_64.tar.gz"`
    For an ARM CPU it's:
    `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.80/coreminer-linux-arm64.tar.gz"`

6. Extract coreminer-linux-x86_64.tar.gz: `tar -xf coreminer-linux-x86_64.tar.gz`
    Or for coreminer-linux-arm64.tar.gz: `tar -xf coreminer-linux-arm64.tar.gz`

7. Change directory: `cd coreapp`

8. Install bc: `sudo apt-get install bc`

9. Start the Coreminer: `bash mine.sh`

For the Core Coin (XCB) Wallet go to: [Wallet Generator](https://github.com/core-coin/wallet-generator)

To go outside of a folder you have to use: `cd ..`

You can stop the miner with 2x **Ctrl+C**. Every time you restart Ubuntu 22.04.2 CLI you just have to change directory `cd coreminer/coreapp/` and start the Coreminer `bash mine.sh`. Also, you can combine them in one command `cd coreminer/coreapp/ && bash mine.sh`.

If you want to change the Coreminer settings you have to make changes in the pool.cfg file in the Coreminer folder. You can open it with e.g. `nano pool.cfg`. To save it hit **Ctrl+X**, **Y**, and then **Enter**. Alternatively, you can also remove `rm pool.cfg` the pool.cfg file and start the Coreminer again.

If there is a new version of the Coreminer ([Releases](https://github.com/catchthatrabbit/coreminer/releases)) navigate outside of the **coreminer** folder and remove it with `rm -r coreminer`. Then repeat this guide but change the version number on step 5 to the newer version or copy the link and paste it between the quotation marks to replace the download link. Use the newer versions at your own risk because you might need to fix some things to make them work.

You don't have to type in every command twice. Just use the arrow keys up and down on your keyboard to navigate through the history of your previously used commands.

To make a software update for the virtual machine use this command: `sudo apt update && sudo apt upgrade -y`.

**Comment by shozo0517** [comment]
I see that only windows 11 is listed, but is it possible to run (mine) on windows 10 as well?

**Reply by tgh90** [reply]
> I see that only windows 11 is listed, but is it possible to run (mine) on windows 10 as well?

As long as you can install Ubuntu LTS from the Microsoft Store you should be able to use the guide.

**Reply by shozo0517** [reply]
Thanks for the reply. I'll give it a try.

**Comment by PacoFi8** [comment]
Windows 11 Coreminer Setup Instructions and Guide for Beginners.
To run your CoreMiner on Windows Ubuntu is needed to run Linux on Windows. 
Search & Download Ubuntu LTS 22.04 (Long Term Support version) from the Microsoft Store at Your start button on your PC.

Open Ubuntu LTS 22.04 version
!!!As a rule, everything that follows after the : needs to be put as a command in Ubuntu to run it!

To Create a new folder put in the command: 
mkdir coreminer-v01960

Change directory: 
cd coreminer-v01960

Download the coreminer software.
For a x86_64 CPU it's: 
wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.60/coreminer-linux-x86_64.zip"

For an ARM CPU it's: 
wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.60/coreminer-linux-arm64.zip"

Install unzip: 
sudo apt-get install unzip

Unzip coreminer-linux-x86_64.zip: 
unzip coreminer-linux-x86_64.zip

Or for coreminer-linux-arm64.zip: 
unzip coreminer-linux-arm64.zip

Start the coreminer: 
bash ./mine.sh

You can see files and folders in your current directory with: 
la
To go outside of a folder you have to use: 
cd ..

You can stop the miner with Ctrl+C and exit the coreminer with Ctrl+C again. Everytime you restart the Ubuntu CLI (Command Line Interface) you just have to change directory (cd coreminer-v01960) and start the coreminer (bash ./mine.sh). If you want to change the coreminer settings you have to make changes in the pool.cfg file in the coreminer-v01960 folder. You can open it with e.g. 'nano pool.cfg'. To save it hit Ctrl+X, Y and then Enter.

I have just redesigned it a bit, ass it was for a lot of people which have no experience in working with UBUNTU what to put into the command field after the :

**Reply by glennsheary** [reply]
hey! im having a bit of trouble. when i enter in the  "unzip coreminer-linux-x86_64.zip" my system say's  "unzip: cannot find or unzip coreminer"  any thoughts?

**Reply by PaulCosteFi8** [reply]
Hi, 
whether you haven't installed the right UBUNtu LTS 22.04 version or you just haven't used the right command which is:

Install unzip:
sudo apt-get install unzip

Unzip coreminer-linux-x86_64.zip:
unzip coreminer-linux-x86_64.zip

Or for coreminer-linux-arm64.zip:
unzip coreminer-linux-arm64.zip

let me know if that helped and works now for you!?

**Comment by Sn4k3E4t3r** [comment]
Hi i get following error on an Raspbian x64 Lite Image:

./coreminer: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found

Has somebody coreminer Running on an Raspberry Pi 4?

**Reply by rastislavcore** [reply]
Raspberry Pi was tested and working, but Raspbian has an older Glibc.
You can use another OS for IoT devices.

**Comment by earld1604** [comment]
Hallo,  what can I do?
Since yesterday this error occurs:”.mine.sh line 97: ./coreminer: cannot execute binary file: Exec format error”.
I didn’t change anything. Before yesterday it startet perfectly.
Can You help me ?
Many thanks, Earl

**Reply by tgh90** [reply]
The first thing I would try is reinstalling the virtual machine. Just remove the Ubuntu LTS and make the steps in this guide again.

**Reply by Kuvala888** [reply]
iam facing the same issue. since yesterday it shows the error :./mine.sh: line 100: ./coreminer: Invalid argument      ....before it was working fine. i reinstalled ubuntu and the coreminer but still the error is coming ???

**Reply by earld1604** [reply]
I was running the miner software version 19.60. Noted on github the latest version, downloaded this latest version 19.63 and this works fine again... happy mining folks!!

**Reply by Kuvala888** [reply]
hey earld1604. thanx. that was the issue. now its working fine. cheers

**Comment by Kuvala888** [comment]
got a dell workstation 7820 with dual cpu. i was mining with 1 cpu (20 core / 40 Thread) with 14kh. today i recieved and installed the second cpu. but the miner is still mining only on one cpu with 40 thread. does anyone a solution how to activate the second cpu in the coreminer ?

**Reply by earld1604** [reply]
Can somebody help/update the previous installation guides, the miner software files are now with extension .tar.gz instead of .zip.  I tried using tar -xvzf command, but not successful.

**Comment by oldDeathman** [comment]
Hi @tgh90, 

please update the descitption, because the both installation urls are mixed :-)

> 5. Download the coreminer software. Current version v0.19.71.
>    For a x86_64 CPU it's: `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.71/coreminer-linux-arm64.tar.gz"`
>    For an ARM CPU it's: `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.71/coreminer-linux-x86_64.tar.gz"`

**Reply by tgh90** [reply]
Thanks, I changed it.

**Comment by glennsheary** [comment]
this is the error message i receive.

I have the coreminer software downloaded from Github. Could there be a
problem with where I store the file?

[image: image.png]

**Reply by tgh90** [reply]
There is no image link.
